### PR TITLE
edgecontroller:  sync node info if node exists when the edge node is registered

### PR DESCRIPTION
**Pull request by zc2638**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

sync node info if node exists when the edge node is registered

**Which issue(s) this PR fixes**:

NONE

**Special notes for your reviewer**:

NONE

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cloud/pkg/edgecontroller/controller/upstream.go [hunk 1]
```diff
@@ -329,6 +329,30 @@ func (uc *UpstreamController) createNode(name string, node *v1.Node) (*v1.Node,
...
+		node.Annotations = make(map[string]string)
+	}
+	for k, v := range current.Annotations {
+		if _, ok := node.Annotations[k]; ok {
+			continue
+		}
+		node.Annotations[k] = v
+	}
+	node.Spec.Taints = append(node.Spec.Taints, current.Spec.Taints...)
+	return uc.kubeClient.CoreV1().Nodes().Update(context.Background(), node, metaV1.UpdateOptions{})
```
[the remark this answers is not in the document]
Comment (zc2638) [reply cloud/pkg/edgecontroller/controller/upstream.go:353]: I need to create node first,but the edge node doesn't exists.Therefore when the edge node is registered, if the node is exists, it needs to synchronize the special labels and others of the edge node.But that's not how it works now

[the remark this answers is not in the document]
Comment (fisherxu) [reply cloud/pkg/edgecontroller/controller/upstream.go:354]: Now we have supported config taints and labels at edge, would you like to auto create node from edge?

[the remark this answers is not in the document]
Comment (zc2638) [reply cloud/pkg/edgecontroller/controller/upstream.go:354]: For scenario requirements, we need to create a node in the cloud first, and then register the edge node. And need some special labels and others for edge nodes

[the remark this answers is not in the document]
Comment (fisherxu) [reply cloud/pkg/edgecontroller/controller/upstream.go:354]: Okey, sounds good, but if we update node the way in this patch, will be the node info sent to edge? And in edgeside, we have get node func, maybe you can get node through it, please take a  look. Ref: https://github.com/kubeedge/kubeedge/blob/master/edge/pkg/metamanager/client/node.go#L58

[the remark this answers is not in the document]
Comment (zc2638) [reply cloud/pkg/edgecontroller/controller/upstream.go:354]: Thank you, but I'm afraid that's not what I need.
I need to control nodes in the cloud.
The service creates the required nodes in the cloud in advance, and the service will only run in the cloud.

1. Pre create node

2. Install edge node and register

3. Synchronization of edge node registration information

When registering the edge node, I don't want to lose the special label of the edge node. I think the synchronization method should be feasible.

I can't wait for the edge node to register to find it. I know it when I need to create it.

[the remark this answers is not in the document]
Comment (zc2638) [reply cloud/pkg/edgecontroller/controller/upstream.go:354]: I created a node in the cloud, but the registration on the edge side updated the status information, and did not update its special label information. I am very distressed.

[the remark this answers is not in the document]
Comment (fisherxu) [reply cloud/pkg/edgecontroller/controller/upstream.go:354]: Get it, I think we need to further discussion for how to implement it(sync node info to edge).

[the remark this answers is not in the document]
Comment (zc2638) [reply cloud/pkg/edgecontroller/controller/upstream.go:354]: Okey, hope to get the solution as soon as possible
Thanks.

**Pull request closed without merging**